rustls/src/conn.rs

```diff
@@ -517,6 +527,16 @@ impl<Data> ConnectionCommon<Data> {
         self.sendable_tls.set_limit(limit);
     }
 
+    /// Sets a limit on the internal buffers used to buffer decoded plaintext.
+    ///
+    /// See [`Self::set_buffer_limit`] for more information on how limits are applied.
+    pub fn set_plaintext_buffer_limit(&mut self, limit: Option<usize>) {
+        self.core
```

Comment:
Could we have a test that observes that this behaves as you want? I would suggest one that firstly observes the "`received plaintext buffer full`" case in default configuration, and then calls this method and observes the error no longer happens. Let me know if you want help with writing that test.